src/Nethermind/Nethermind.Core/Extensions/SpanExtensions.cs

```diff
@@ -176,48 +176,49 @@ public static ArrayPoolList<T> ToPooledList<T>(this in ReadOnlySpan<T> span)
         [SkipLocalsInit]
         public static int FastHash(this ReadOnlySpan<byte> input)
         {
+            // Very fast hardware accelerated non-cryptographic hash function
             var length = input.Length;
             if (length == 0) return 0;
 
             ref var b = ref MemoryMarshal.GetReference(input);
+            uint hash = s_instanceRandom + (uint)length;
+            if (length < sizeof(long))
+            {
+                goto Short;
+            }
 
-            // Start with instance random, length and first as seed
-            uint hash = (s_instanceRandom + (uint)length) ^ b;
-
-            // This is done below, without branches
-            // if ((length & 1) == 1)
-            // {
-            //     hash = b;
-            //     b = ref Unsafe.Add(ref b, 1);
-            //     length -= 1;
-            // }
-
-            uint bit = (uint)length & sizeof(byte);
-            b = ref Unsafe.Add(ref b, bit);
-            length -= (int)bit;
-
-            if ((length & sizeof(ushort)) != 0)
+            // Start with instance random, length and first ulong as seed
+            hash = BitOperations.Crc32C(hash, Unsafe.ReadUnaligned<ulong>(ref b));
+            if ((length & sizeof(ulong)) != 0)
```

Comment:
@benaadams I believe this is not correct because if the length is bigger than 8, you never align to length & 7.

Right after reading the first long, 

```
                // Skip the bytes that don't align to ulong
                uint bits = (uint)length & 7;
                b = ref Unsafe.Add(ref b, bits);
                length -= (int)bits;
```

should follow in a branchless manner